cmd/podman/containers/podmansh.go

```diff
@@ -0,0 +1,77 @@
+package containers
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/libpod/define"
+	"github.com/spf13/cobra"
+)
+
+func podmanshWait(args []string) error {
+	maxDuration := 10 * time.Second
+	interval := 100 * time.Millisecond
+	ticker := time.Tick(interval)
+
+	cond, err := define.StringToContainerStatus("running")
+	if err != nil {
+		return err
+	}
+	waitOptions.Condition = append(waitOptions.Condition, cond)
+
+	startTime := time.Now()
+	for time.Since(startTime) < maxDuration {
+		<-ticker
+		response, _ := registry.ContainerEngine().ContainerWait(context.Background(), args, waitOptions)
+		if len(response) != 1 {
```

Comment:
@rhatdan we do need this for the case where we run it as a shell command. Otherwise podmansh segfaults if the container isn't running. If that situation is never expected to occur, I guess I can delete it, but there will be the annoying segfault everytime one runs it locally.